framework/src/meshgenerators/StitchedMeshGenerator.C

```diff
@@ -129,6 +134,40 @@ StitchedMeshGenerator::generate()
 
     const bool use_binary_search = (_algorithm == "BINARY");
 
+    // Avoid chaotic boundary merging due to overlapping ids in meshes by simply renumbering the
+    // boundaries in the meshes we are going to stitch onto the base mesh
+    // This is only done if there is an overlap
+    if (_prevent_boundary_ids_overlap)
+    {
+      // Meshes must be prepared to get the global boundary ids and global boundary ids
+      // must be used in parallel to be sure to renumber boundaries consistently across processes
+      if (!mesh->is_prepared())
+        mesh->prepare_for_use();
+      if (!meshes[i]->is_prepared())
+        meshes[i]->prepare_for_use();
```

Comment:
this could be real expensive but there's no choice
https://mooseframework.inl.gov/docs/doxygen/libmesh/classlibMesh_1_1BoundaryInfo.html#a2a6636386e620473deb0f30eb1f996bc

Reply:
I introduced this stuff in libmesh/libmesh@975c67997daff8d85629301a39f2f282e0390440. I was thinking whether we should relax that assertion, but probably not. I don't know whether @roystgnr would have an opinion.

Anyway I am fine with these lines